### PR TITLE
New Provisioning API

### PR DESCRIPTION
**Pull request by satlead**

### CODE REVIEW

**Review by jlacivita — CHANGES_REQUESTED** [review]

File: src/diagrams/provision.puml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+group OS Bootup
+    boot -> device: check if device is provisioned
+    device --> rdk : get device provision info
+    rdk -> boot : provision details
+    group Not Provisioned
+        boot -> ui :launch
+        ui -> core:Authentication.token("device")
+        core -> ui:Device Token
+        ui -> core:Device.id()
```
Comment (jlacivita) [inline src/diagrams/provision.puml:25]: I think `Device.id` returns the back-office ID, not the hardware MAC address?

File: src/diagrams/provision.puml [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        core -> ui:Mac Address
+        ui -> distributor:Provision
+        note right of ui: Distributor uses the sensitive device info \nto setup billing and services.
+        distributor -> ui:Account Id, Device Id, Distributor Id
```
Comment (jlacivita) [inline src/diagrams/provision.puml:29]: Once we settle on which parameters are in the API, update this arrow to match.

**Review by jlacivita — COMMENTED** [review]
Will let @kpears201 approve if this API is a comprehensive solution. I made some comments on the PUML.

**Review by kpears201 — APPROVED** [review]

File: src/diagrams/provision.puml [hunk 3]
```diff
@@ -0,0 +1,54 @@
+@startuml
+
+box os
+    participant boot as "Boot"
```
[the remark this answers is not in the document]
Comment (satlead) [reply src/diagrams/provision.puml]: This file is moved as it was not directly applying to the SDK

**Pull request merged**